src/System.Management.Automation/engine/AsyncByteStreamDrainer.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Management.Automation;
+
+internal sealed class AsyncByteStreamDrainer : IDisposable
```

Comment:
Really don't like this name. `StreamReader` implies string or I'd use that. Open to suggestions

Reply:
AsyncByteStreamRepeater?

Reply:
`AsyncByteStreamTransfer` maybe?